src/Maker/MakeAuthenticator.php

```diff
@@ -55,50 +70,94 @@ public function configureCommand(Command $command, InputConfiguration $inputConf
+            function ($answer) {
+                Validator::notBlank($answer);
+
+                return Validator::classDoesNotExist(
+                    $this->generator->createClassNameDetails(
+                        $answer,
+                        'Security\\'
+                    )->getFullName()
+                );
+            }
         );
+        $input->setArgument('authenticator-class', $io->askQuestion($questionAuthenticatorClass));
 
+        $command->addOption('firewall-name', null, InputOption::VALUE_OPTIONAL);
+        $input->setOption('firewall-name', $firewallName = InteractiveSecurityHelper::guessFirewallName($io, $securityData));
+
+        $command->addOption('entry-point', null, InputOption::VALUE_OPTIONAL);
         $input->setOption(
             'entry-point',
-            $interactiveSecurityHelper->guessEntryPoint($io, $securityData, $authenticatorClassNameDetails->getFullName(), $firewallName)
+            InteractiveSecurityHelper::guessEntryPoint($io, $securityData, $input->getArgument('authenticator-class'), $firewallName)
         );
+
+        if (self::AUTH_TYPE_FORM_LOGIN === $input->getArgument('authenticator-type')) {
+            $command->addArgument('controller-class', InputArgument::OPTIONAL);
+            $input->setArgument(
+                'controller-class',
+                $io->ask(
+                    'Choose a name for the controller class (e.g. <fg=yellow>SecurityController</>)',
+                    'SecurityController',
+                    [Validator::class, 'validateClassName']
+                )
+            );
+
+            $command->addArgument('user-class', InputArgument::OPTIONAL);
+            $input->setArgument('user-class', InteractiveSecurityHelper::guessUserClass($io, $securityData));
+        }
     }
 
     public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator)
     {
-        $classNameDetails = $generator->createClassNameDetails(
-            $input->getArgument('authenticator-class'),
-            'Security\\'
-        );
-
+        // generate authenticator class
         $generator->generateClass(
-            $classNameDetails->getFullName(),
-            'authenticator/Empty.tpl.php',
-            []
+            $input->getArgument('authenticator-class'),
+            self::AUTH_TYPE_FORM_LOGIN === $input->getArgument('authenticator-type') ?
+                ($this->doctrineHelper->isClassAMappedEntity($input->getArgument('user-class')) ? 'authenticator/LoginFormEntityAuthenticator.tpl.php' : 'authenticator/LoginFormNotEntityAuthenticator.tpl.php')
+                : 'authenticator/Empty.tpl.php',
+            $input->hasArgument('user-class') ?
+                [
+                    'user_fully_qualified_class_name' => $input->getArgument('user-class'),
+                    'user_class_name' => substr(strrchr($input->getArgument('user-class'), '\\'), 1),
+                ]
+                : []
```

Comment:
I found the ternary you were asking about :p. I think an if statement around this entire block is much more clear. Actually, even a switch-case

```php
switch ($input->getArgument('authenticator-type')) {
    case self::AUTH_TYPE_FORM_LOGIN:
        $generator->generateClass(
            // ...
        );
        break;
    case self::self::AUTH_TYPE_EMPTY_AUTHENTICATOR:
        $generator->generateClass(
            // ...
        );
        break;
    default:
        throw new \Exception('Invalid authenticator type');
}
```

Reply:
ok that's fair ;)